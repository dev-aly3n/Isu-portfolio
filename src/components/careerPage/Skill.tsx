interface props {
  skill: { name: string; icon: string; perc: number; color: string };
  index: number;
}

const Skill: React.FC<props> = ({ skill, index }) => {
  return (
    <div className="flex justify-center items-center relative bg-gray-600 rounded-full w-[88%] md:w-[45%] h-12 
    self-start justify-self-start place-self-start">
      <h3 className="w-32 h-full bg-gray-800 flex justify-center items-center rounded-l-full text-xs font-bold break-words text-center">
        {skill.name}
      </h3>
      <div className="relative flex w-full h-full bg-gray-500 rounded-r-full">
        <div
          className="absolute top-[50%] mt-[-10px] left-0  h-5 overflow-hidden rounded-r-full pos-dash-style "
          style={{
            width: `${skill.perc - 7}%`,
            backgroundColor: `rgb${skill.color}`,
          }}
        ></div>
        <img
          src={`/media/logos/${skill.icon}`}
          style={{ left: `${skill.perc - 7}%` }}
          className="absolute top-[50%] mt-[-14px] block h-7 ml-px"
        />
        {skill.name === "Creativity" && (
          <img
            src={`/media/logos/explosion.png`}
            style={{ left: `100%` }}
            className="absolute top-[50%] mt-[-30px] block "
          />
        )}
      </div>
    </div>
  );
};

export default Skill;
