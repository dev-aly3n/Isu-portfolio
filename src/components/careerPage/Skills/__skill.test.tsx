import { render, screen } from "@testing-library/react";
import Skill from "./Skill";
import { skillMock } from "../../../__mock__";

describe("career/Skill", () => {
  it("render name and img", () => {
    render(<Skill skill={skillMock} />);
    const skillName = screen.getByTestId('heading');
    const skillImg = screen.getByTestId('skill-img');
    expect(skillName).toBeInTheDocument();
    expect(skillImg).toBeInTheDocument();
  });
});
