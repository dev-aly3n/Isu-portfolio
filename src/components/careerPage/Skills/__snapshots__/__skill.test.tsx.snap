// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`career/Skill skill snap 1`] = `
<div>
  <div
    class="skill-container"
  >
    <h3
      data-testid="heading"
    >
      Javascript
    </h3>
    <div>
      <div
        style="background-color: rgb(247, 223, 30);"
      />
      <span
        style="left: 68%;"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Javascript"
            data-nimg="fill"
            data-testid="skill-img"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </span>
      </span>
    </div>
  </div>
</div>
`;
